media-gfx/freecad/freecad-1.0_rc2-r1.ebuild

```diff
@@ -294,12 +295,16 @@ src_configure() {
 # configuration. Without those, there are sandbox violation, when it
 # tries to create /var/lib/portage/home/.FreeCAD directory.
 src_test() {
-	pushd "${BUILD_DIR}" > /dev/null || die
-	export FREECAD_USER_HOME="${HOME}"
-	export FREECAD_USER_DATA="${T}"
-	export FREECAD_USER_TEMP="${T}"
-	nonfatal ./bin/FreeCADCmd --run-test 0
-	popd > /dev/null || die
+	(
+		pushd "${BUILD_DIR}" > /dev/null || die
+		export FREECAD_USER_HOME="${HOME}"
+		export FREECAD_USER_DATA="${T}"
+		export FREECAD_USER_TEMP="${T}"
+		nonfatal ./bin/FreeCADCmd --run-test 0
+		ret=${?}
+		popd > /dev/null || die
+		exit "${ret}"
```

Comment:
Why was nonfatal ever used here at all? `bin/FreeCADCmd` is not an ebuild helper, it doesn't internally call `die` on failure?

Fixing this by saving `$ret` and dying at the end after popd isn't great. There is no point in guaranteeing popd runs even when dying. In fact you should go out of your way to avoid popd, since portage will print a stack trace on failure which includes the directory the error occurred in -- and it is quite helpful for portage to be able to report the directory the tests occurred in, rather than `${S}`.

See for example dbeb507f468dcc75095d54b59c165dcc70ef558d.

Reply:
> So, assuming that is OK, what to do - add a new commit? Replace [ecfdfed](https://github.com/gentoo/gentoo/commit/ecfdfed57ec94f2760050537c760a955614130a3)?

Any time a new commit involves a change of heart about a previous commit in the same series, it should replace the previous commit rather than do both in turn.

This means that you don't end up with a final history where you make an unwanted change, just to then delete that change and make a different change instead.